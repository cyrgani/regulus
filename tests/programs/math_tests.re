assert_eq(%(5, 3), 2),
assert_eq(%(3, 5), 3),

import(math),

assert_eq(abs(-1), abs(1)),
assert_eq(abs(-0), abs(0)),
assert_eq(abs(-2), 2),
assert_eq(abs(-9223372036854775807), 9223372036854775807),
__builtin_print_catch(abs(-9223372036854775808)),

assert_eq(gcd(5, 1), 1),
assert_eq(gcd(5, 3), 1),
assert_eq(gcd(16, 4), 4),
assert_eq(gcd(13, 7), 1),
assert_eq(gcd(30, 12), 6),
assert_eq(gcd(12, 30), 6),
__builtin_print_catch(gcd(0, 8)),
__builtin_print_catch(gcd(8, 0)),
__builtin_print_catch(gcd(0, 0)),
assert_eq(gcd(8, 8), 8),
# TODO: test negative numbers
