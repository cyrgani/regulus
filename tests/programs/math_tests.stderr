OverflowError: overflow occured during -
at <stl:operators>:35:26
at <stl:operators>:33:11
at <stl:operators>:30:19
at <stl:math>:2:34
at <stl:math>:2:19
at programs/math_tests.re:10:26
at programs/math_tests.re:10:22
DivideByZeroError: cannot calculate gcd when one argument is zero
at <stl:math>:7:37
at <stl:control_flow>:66:39
at <stl:control_flow>:66:28
at <stl:math>:7:7
at <stl:math>:6:17
at programs/math_tests.re:18:26
at programs/math_tests.re:18:22
DivideByZeroError: cannot calculate gcd when one argument is zero
at <stl:math>:7:37
at <stl:control_flow>:66:39
at <stl:control_flow>:66:28
at <stl:math>:7:7
at <stl:math>:6:17
at programs/math_tests.re:19:26
at programs/math_tests.re:19:22
DivideByZeroError: cannot calculate gcd when one argument is zero
at <stl:math>:7:37
at <stl:control_flow>:66:39
at <stl:control_flow>:66:28
at <stl:math>:7:7
at <stl:math>:6:17
at programs/math_tests.re:20:26
at programs/math_tests.re:20:22
